src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -107,6 +107,55 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
     }
   }
 
+  // Override updateWith for performance, so we can do the update while hashing
+  // the input key only once and performing one lookup into the hash table
+  override def updateWith(key: K)(remappingFunction: Option[V] => Option[V]): Option[V] = {
```

Comment:
This implementation should probably all be inside an 

```scala
  if (getClass != classOf[HashMap[_, _]]) {
      // subclasses of HashMap might customise `get` ...
      super.getOrElse(key, default)
    } else {
      ...
   }
```

Since if a subclass overrides `get` or `update` then `updateWith` may no longer be doing the right thing. (see `getOrElse`/`getOrElseUpdate`)

Reply:
Done